internal/providers/pluginfw/resources/catalog/resource_function.go

```diff
@@ -0,0 +1,226 @@
+		c.SetRequired("name")
+		c.SetRequired("catalog_name")
+		c.SetRequired("schema_name")
+		c.SetRequired("input_params")
+		c.SetRequired("data_type")
+		c.SetRequired("routine_body")
+		c.SetRequired("routine_defintion")
+		c.SetRequired("language")
+
+		c.SetReadOnly("full_name")
+		c.SetReadOnly("created_at")
+		c.SetReadOnly("created_by")
+		c.SetReadOnly("updated_at")
+		c.SetReadOnly("updated_by")
+
+		return c
+	})
+
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Function",
+		Attributes:  attrs,
+		Blocks:      blocks,
+	}
+}
+
+func (r *FunctionResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if r.Client == nil && req.ProviderData != nil {
+		r.Client = pluginfwcommon.ConfigureResource(req, resp)
+	}
+}
+
+func (r *FunctionResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("full_name"), req, resp)
+}
+
+func (r *FunctionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var planFunc catalog_tf.FunctionInfo
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planFunc)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createReq catalog.CreateFunctionRequest
+
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, planFunc, &createReq)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	funcInfo, err := w.Functions.Create(ctx, createReq)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create function", err.Error())
+	}
+
+	resp.Diagnostics.Append(waitForFunction(ctx, w, funcInfo)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, funcInfo, &planFunc)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, planFunc)...)
+}
+
+func (r *FunctionResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var planFunc catalog_tf.FunctionInfo
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planFunc)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var updateReq catalog.UpdateFunction
+
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, planFunc, &updateReq)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	funcInfo, err := w.Functions.Update(ctx, updateReq)
```

Comment:
is waiting only required during creation? or also during update?

Reply:
Removing the waiting process for functions, as they do not depend on any other resource and thus should be created instantly.